warp/Network/Wai/Handler/Warp/Settings.hs

```diff
@@ -249,7 +249,16 @@ exceptionResponseForDebug e =
 -- @since 3.3.17
 defaultFork :: ((forall a. IO a -> IO a) -> IO ()) -> IO ()
 defaultFork io =
+#if __GLASGOW_HASKELL__ >= 904
```

Comment:
Would you provide us a material to know the background of this change?

Reply:
Would love to but could not find any info on the changelog for `base` or `ghc-prim` about this. When GHC 9.4 is closer to release, I hope that this info will be more obvious.

Reply:
@snoyberg I trust Matt and have no objection for merging. What about you?

Reply:
No objection from me either, go for it! As always thanks Matt 💪

Reply:
Of course!

True 9.4 support relies on updated `foundation` and `basement`, but I think we can release this as-is and it'll work when those libraries are up-to-date.

I can leave the `cabal.project` in the repo and merge as-is, or we can leave pending until that's been taken care of.